### PR TITLE
create open_port function to automatically open ports over upnp

### PR DESCRIPTION
**Pull request by enoch85**
New PR for https://github.com/nextcloud/vm/pull/1464

### CODE REVIEW

**Comment by szaimen** [comment]
Why do we need this? it seems like https://github.com/nextcloud/vm/pull/1464 was already merged into master?

**Comment by enoch85** [comment]
> Why do we need this? it seems like #1464 was already merged into master?

Screenshot from 2020-09-17 12:08:33

**Review by szaimen — COMMENTED** [review]
Hm... I cant delete this review...

File: lib.sh [hunk 1]
```diff
@@ -698,6 +698,26 @@ fi
...
+    fi
+}
+
+cleanup_open_port() {
+    if [ -n "$FAIL" ]
+    then
+        apt-get purge miniupnpc -y
+        apt autoremove -y
+    fi
+}
```
Comment (szaimen) [inline lib.sh:719]: I mean the solution with cleanup_open_port and unset FAIL?

File: apps/talk_signaling.sh [hunk 2]
```diff
@@ -303,14 +320,6 @@ check_command systemctl enable signaling
 # Apache Proxy
 # https://github.com/strukturag/nextcloud-spreed-signaling#apache
 
-# Check if $SUBDOMAIN exists and is reachable
-print_text_in_color "$ICyan" "Checking if $SUBDOMAIN exists and is reachable..."
-domain_check_200 "$SUBDOMAIN"
-
-# Check open ports with NMAP
-check_open_port 80 "$SUBDOMAIN"
```
Comment (enoch85) [inline apps/talk_signaling.sh:311]: why are these removed?

Comment (enoch85) [reply apps/talk_signaling.sh:311]: Sorry, they are moved.

File: lib.sh (hunk 1 above)
Comment (enoch85) [reply lib.sh:719]: Looks good to me. :+1:

Comment (enoch85) [reply lib.sh:719]: But I think thet could be in the same funtion. I will fix.

Comment (enoch85) [reply lib.sh:719]: Actually, no you're right, this way is better.

**Comment by enoch85** [comment]
> Hm... I cant delete or hide this comment...

Report the bug to Github :laughing:

**Comment by szaimen** [comment]
@enoch85 this looks strange to me:

**Comment by enoch85** [comment]
No, it's correct. The port check happens after the try to use UPNP.

**Comment by szaimen** [comment]
> No, it's correct. The port check happens after the try to use UPNP - which is after you hit OK.

You didn't get what I meant: it is not indide the msg_box?
